install/ui/src/freeipa/Application_controller.js

```diff
@@ -255,13 +261,36 @@ define([
             return self_service;
         },
 
-        update_logged_in: function(logged_in, fullname) {
+        is_aduser_selfservice: function() {
+            var selfservice = IPA.whoami.metadata.object === 'idoverrideuser';
+            // quite ugly, needed for users and iduseroverride to hide breadcrumb
+            IPA.is_aduser_selfservice = selfservice;
+
+            return selfservice;
+        },
+
+        update_logged_in: function(logged_in) {
             this.app_widget.set('logged', logged_in);
+
+            var whoami = IPA.whoami;
+            var fullname = '';
+            var entity = whoami.metadata.object;
+
+            if (entity === 'user') fullname = whoami.data.krbprincipalname[0];
```

Comment:
I can call `user-show login` with all and change the behavior to try to use `cn` and in case that there is no `cn` I can use `krbprincipalname`. Would it be better?